pkg/cloud/ovirt/clients/machineservice.go

```diff
@@ -423,3 +458,77 @@ func (is *InstanceService) handleAffinityGroups(vm *ovirtsdk.Vm, cID string, ags
 	}
 	return nil
 }
+
+// getHostsInCluster will return a HostSlice of the hosts in a given cluster ID.
+func (is *InstanceService) getHostsInCluster(clusterId string) (*ovirtsdk.HostSlice, error) {
+	clusterService := is.Connection.SystemService().ClustersService().ClusterService(clusterId)
+	clusterGet, err := clusterService.Get().Send()
+	if err != nil {
+		return nil, errors.Errorf("failed to get the cluster: %v", err)
+	}
+	clusterName := clusterGet.MustCluster().MustName()
+	hostsInCluster, err := is.Connection.SystemService().HostsService().List().Search(
+		fmt.Sprintf("cluster=%s", clusterName)).Send()
+	if err != nil {
+		return nil, errors.Errorf("failed to get the list of hosts in the cluster: %v", err)
+	}
+	return hostsInCluster.MustHosts(), nil
+}
+
+// handleAutoPinning updates the VM after creation to set the auto pinning policy configuration.
+func (is *InstanceService) handleAutoPinning(id string, autoPinningPolicy ovirtsdk.AutoPinningPolicy) error {
+	vmService := is.Connection.SystemService().VmsService().VmService(id)
+	optimizeCpuSettings := !(autoPinningPolicy == ovirtsdk.AUTOPINNINGPOLICY_EXISTING)
+	_, err := vmService.AutoPinCpuAndNumaNodes().OptimizeCpuSettings(optimizeCpuSettings).Send()
+	if err != nil {
+		return errors.Errorf("failed to set the auto pinning policy on the VM!, %v", err)
+	}
+	return nil
+}
+
+// getEngineVersion will return the engine version we are using
+func (is *InstanceService) getEngineVersion() *ovirtsdk.Version {
+	return is.Connection.SystemService().Get().MustSend().MustApi().MustProductInfo().MustVersion()
+}
+
+// versionCompare will take two *ovirtsdk.Version and compare the two
+func versionCompare(v *ovirtsdk.Version, other *ovirtsdk.Version) (int64, error) {
+	if v == nil || other == nil {
+		return 5, fmt.Errorf("can't compare nil objects")
+	}
+	if v == other {
+		return 0, nil
+	}
+	result := v.MustMajor() - other.MustMajor()
+	if result == 0 {
+		result = v.MustMinor() - other.MustMinor()
+		if result == 0 {
+			result = v.MustBuild() - other.MustBuild()
+			if result == 0 {
+				result = v.MustRevision() - other.MustRevision()
+			}
+		}
+	}
+	return result, nil
+}
+
+// autoPinningSupported will check if the engine's version is relevant for the feature.
+func (is *InstanceService) autoPinningSupported() (bool, error) {
+	engineVer := is.getEngineVersion()
+	autoPiningRequiredEngineVersion := ovirtsdk.NewVersionBuilder().
+		Major(4).
+		Minor(4).
+		Build_(5).
+		Revision(0).
+		MustBuild()
+	versionCompareResult, err := versionCompare(engineVer, autoPiningRequiredEngineVersion)
+	if err != nil {
+		return false, err
+	}
+	if versionCompareResult >= 0 {
+		return true, nil
+	}
+	klog.Warning("The engine version %d.%d.%d is not supporting the auto pinning feature. " +
+		"Please update to 4.4.5 or later.", engineVer.MustMajor(), engineVer.MustMinor(), engineVer.MustBuild())
+	return false, nil
```

Comment:
I think you should return an error and not a warning in the logs here,
In the current case things will not happen and there will be some hidden log in a place no one is looking at...
After https://github.com/openshift/cluster-api-provider-ovirt/pull/96 we should port that part of the version check to the validateMachine function and return an InvalidMachineConfiguration error

Reply:
The PR got merges so we should move this check to the validations